include/boost/url/impl/scheme.ipp

```diff
@@ -107,6 +107,28 @@ to_string(scheme s) noexcept
     return "<unknown>";
 }
 
+std::uint16_t
+default_port(scheme s) noexcept
+{
+    switch(s)
+    {
+    case scheme::ftp:
+        return 21;
+    case scheme::http:
+    case scheme::ws:
+        return 80;
+    case scheme::https:
+    case scheme::wss:
+        return 443;
+    case scheme::file:
+    case scheme::none:
+        return -1;
```

Comment:
You should assert that the scheme is not `none` and you should treat `unknown` as 0

Reply:
Mmm... Yes. `0` is much better since it's invalid.  

My ideia wasn't very smart. I thought `-1` would be kind of an `npos` for the port. But that's not a good idea because the literal `-1` won't be equal to that.